text/0033-node-event-api/0033-node-event-api.yml

```diff
@@ -0,0 +1,163 @@
+asyncapi: 2.0.0
+info:
+  title: Node Event API
+  contact:
+    email: contact@iota.org
+  version: 1.0.0
+  description: The node event API is in charge of publishing information about events within the node software.
+externalDocs:
+  description: Find out more about IOTA
+  url: 'https://iota.org'
+channels:
+  milestones/latest:
+    subscribe:
+      operationId: milestoneLatest
+      description: Publishes the newest latest known milestone.
+      message:
+        $ref: '#/components/messages/MilestonePayload'
+  milestones/solid:
+    subscribe:
+      operationId: milestoneSolid
+      description: Publishes the newly solid milestone.
+      message:
+        $ref: '#/components/messages/MilestonePayload'
+  messages:
+    subscribe:
+      operationId: messages
+      description: Publishes newly received messages in their serialized binary form.
+      message:
+        $ref: '#/components/messages/Message'
+  messages/referenced:
+    subscribe:
+      operationId: messagesReferenced
+      description: Publishes metadata of messages newly referenced by a milestone.
+      message:
+        $ref: '#/components/messages/MessageMetadata'

```

Comment:
Do we really need all these fields for the `messages/referenced` endpoint? Or could we only include:
- messageId
- milestoneIndex
- ledgerInclusionState
- (maybe is_solid?)

and if other information is needed, the `metadata` endpoint can be subscribed for more information?

Reply:
I don't think it makes sense to truncate the information because then you have to use multiple channels to get to your perhaps wanted data. If you look at the other fields such as `shouldPromote`/`shouldReattach` you can see that they are optional and in fact not added if the given msg already has been referenced.